core/include/scipp/core/element/arithmetic.h

```diff
@@ -71,26 +71,42 @@ constexpr auto mod_equals = overloaded{
 
 struct add_types_t {
   constexpr void operator()() const noexcept;
-  using types = arithmetic_and_matrix_type_pairs;
+  using types =
+      decltype(std::tuple_cat(std::declval<arithmetic_and_matrix_type_pairs>(),
+                              std::tuple<std::tuple<time_point, int64_t>,
+                                         std::tuple<int64_t, time_point>>{}));
+};
+
+struct minus_types_t {
+  constexpr void operator()() const noexcept;
+  using types = decltype(std::tuple_cat(
+      std::declval<arithmetic_and_matrix_type_pairs>(),
+      std::tuple<std::tuple<time_point, int64_t>,
+                 std::tuple<time_point, time_point>>{}));
 };
 
 struct times_types_t {
   constexpr void operator()() const noexcept;
   using types = decltype(std::tuple_cat(
       std::declval<arithmetic_type_pairs_with_bool>(),
-      std::tuple<std::tuple<Eigen::Matrix3d, Eigen::Matrix3d>>(),
-      std::tuple<std::tuple<Eigen::Matrix3d, Eigen::Vector3d>>()));
+      std::tuple<std::tuple<Eigen::Matrix3d, Eigen::Matrix3d>>{},
+      std::tuple<std::tuple<Eigen::Matrix3d, Eigen::Vector3d>>{},
+      std::tuple<std::tuple<time_point, int64_t>,
+          std::tuple<int64_t, time_point>>{}));
```

Comment:
This does not feel right? Times should not be scalable. What would the interpretation of `17 * 9:15 am` be?

Reply:
Right. I added them to allow for unit conversions. But those can be done differently. Removed it now.

Reply:
Here is a case that is not possible without the operator:
```.py
np.datetime64(123, 's') * sc.units.s
```

Reply:
I don't understand, why should this multiply two variables? It is just a shorthand for constructing variables?

Reply:
But it multiplies the underlying `core::time_point` and `int`

Reply:
Which integer? There is no integer, just a unit.

Reply:
It's working now